adal/authority.py

```diff
@@ -63,9 +64,15 @@ def url(self):
         return self._url.geturl()
 
     def _whitelisted(self): # testing if self._url.hostname is a dsts whitelisted domain
-        for domain in AADConstants.WHITELISTED_DOMAINS:
+        for domain in self._whitelisted_domains:
             if self._url.hostname.endswith(domain):
                 return True
+        # Add dSTS domains to whitelist based on based on domain
+        # https://microsoft.sharepoint.com/teams/AzureSecurityCompliance/Security/SitePages/dSTS%20Fundamentals.aspx
+        if self._url.hostname.find(".dsts.") > -1:

```

Comment:
If this is the real check you are going to require, how about rewrite this entire helper as below?

```python
def _whitelisted(self):
    # Add dSTS domains to whitelist based on based on domain
    # https://microsoft.sharepoint.com/teams/AzureSecurityCompliance/Security/SitePages/dSTS%20Fundamentals.aspx
    return ".dsts." in self._url.hostname
```

(and then also remove the [new, now unnecessary, line 46](https://github.com/AzureAD/azure-activedirectory-library-for-python/pull/215/files#diff-03f44f824a1959ef23cc5f4861b150a5R46))?